src/text_buffer.rs

```diff
@@ -486,6 +495,8 @@ impl TextBuffer {
     }
 
     pub fn save(&mut self) -> std::result::Result<String, String> {
+        self.insert_undo_point();
```

Comment:
Why is this necessary? Text is not modified on saving file.

And by this line, saving files multiple times without modifying text inserts multiple undo points.

Reply:
It happens that `save` function is called but some ongoing changes exist.
I want to put these changes into history at the time,
so the text buffer is unmodified in saved state.

It seems no undo point inserted calling `insert_undo_point` function multiple times
because `self.inserted_undo` is `false`.

Reply:
I got the point. Thanks for explanation. However, inserting undo point here means that an undo point is created even if saving file is cancelled. Can we move this line just before returning from this function?

Reply:
OK, I got the point. Thanks for the explanation. I tried this change on my local and it worked fine.